tests/check/check.go

```diff
@@ -125,63 +133,87 @@ func runTests(t *testing.T, schemaVersion string, client ClientInterface) {
 	err = yaml.Unmarshal(b, &testCases)
 	require.NoError(t, err)
 
-	ctx := context.Background()
-
 	for _, test := range testCases.Tests {
 		test := test
+		runTest(t, test, testInformation, false)
+		runTest(t, test, testInformation, true)
 
-		t.Run(test.Name, func(t *testing.T) {
-			t.Parallel()
-			resp, err := client.CreateStore(ctx, &openfgapb.CreateStoreRequest{Name: test.Name})
-			require.NoError(t, err)
+	}
+}
+
+func runTest(t *testing.T, test individualTest, testInformation testInformation, contextTupleTest bool) {
+	schemaVersion := testInformation.schemaVersion
+	client := testInformation.client
+	name := test.Name
+	if contextTupleTest {
+		name += "_ctxTuples"
+	}
 
-			storeID := resp.GetId()
+	if contextTupleTest && len(test.Stages) > 1 {
+		// we don't want to run special contextual tuples test for these cases
+		// as multi-stages test has expectation tuples are in system
+		return
```

Comment:
```suggestion
	if contextTupleTest {
		name += "_ctxTuples"
	}

	t.Run(name, func(t *testing.T) {
		if contextTupleTest && len(test.Stages) > 1 {
			t.Skipf("multi-stages test has expectation tuples are in system")
		}
```

Reply:
I don't understand why you marked this as resolved? My proposal involves skipping that test

Reply:
added change